### DISCUSSION TITLE
Mask the object and fill the background with white

### DISCUSSION THREAD

**Discussion by TNNSLAB** [ImageMagick/ImageMagick#6281] [Help]
### ImageMagick version

Version: ImageMagick 7.1.0-4 Q16 x86_64 2021-07-18

### Operating system, version and so on

Mac OS X 10.14.6

### Description

How do I make a good selection around this kind of objects and mask everything around with white ? 

![full_2](https://user-images.githubusercontent.com/46636486/234272176-5aa644de-0c2a-48d3-9c0b-8400edb8ce16.jpg)
I need :
![full_2 copy](https://user-images.githubusercontent.com/46636486/234272997-b480f7bd-281b-40c4-a3f6-e6adc98c9482.jpg)

**Comment by fmw42** [comment]
The white on the shoes is too close to the white background to be able to separate the shoe from the background as one foreground object for the shoe for simple image processing to handle consistently.  I suggest you use AI/Deep Learning, for example http://remove.bg.  It does a good job on this image.

**Comment by TNNSLAB** [comment]
@fmw42 what if i have this kind of contour around it ? 
![result2](https://user-images.githubusercontent.com/46636486/234513268-68af0b72-290f-4fc7-98f5-36eccfb1355d.jpg)

**Comment by fmw42** [comment]
```
Line 1: read the input and put into MPR: for later
Line 2: make anything not green into white and anything not white into black
Line 3: flood fill with black from the top left corner up to the black outline to create a mask
Line 4: make a white image
Line 5: bring back the MPR:img and reverse the order of all 3 images
Line 6: composite the input with the white using the mask
Line 7: change remnant green to white as best can be done without losing the object
Line 8: save the result

```

Input:

![234513268-68af0b72-290f-4fc7-98f5-36eccfb1355d](https://user-images.githubusercontent.com/13458218/234632721-7c9850d7-1a8d-41d2-a73e-8c6bbc3c3eeb.jpg)



```
convert 234513268-68af0b72-290f-4fc7-98f5-36eccfb1355d.jpg -write mpr:img \
-fuzz 15% -fill white +opaque "rgb(14,252,18)" -fill black +opaque white \
-fill black -draw "color 0,0 floodfill" -alpha off -negate \
\( mpr:img -fill white -colorize 100 \) \
mpr:img -reverse \
-compose over -composite \
-fuzz 25% -fill white -opaque "rgb(14,252,18)" \
result.jpg

```

![result](https://user-images.githubusercontent.com/13458218/234632757-c840d0e8-6e4c-4489-8b5d-c122b45a5e04.jpg)

**Reply by TNNSLAB** [reply]
That looks very promising is there a way to get rid of green contour afterwards ?

**Reply by fmw42** [reply]
You drew anti-aliased green line. You are seeing the anti-aliased part that I could not remove with -fuzz.  If you draw solid green with no anti-aliasing (no smoothing), it could all be removed.
